src/log.h

```diff
@@ -93,7 +93,7 @@ typedef int UDS_LogLevel_t;
 #define UDS_LOG_SDU(tag, buffer, buff_len, info) UDS_LogSDUDummy(tag, buffer, buff_len, info)
 #endif
 
-#if defined(__GNUC__) || defined(__clang__)
+#if (defined(__GNUC__) || defined(__clang__)) && !defined(UDS_SYS_RTT)
```

Comment:
what is the reason to disable ` __attribute__((format(...)))` type checking when using RT-Thread?

Reply:
- Because RTT has its own printf function, and it also includes the confirmation of the format. In the RTT environment, not enabling it will result in a compilation error.